### DISCUSSION TITLE
Custom endpoint & authorization header = INVALID_CREDENTIALS

### DISCUSSION THREAD

**Discussion by fusscreme** [directus/directus#17595] [Archive]
Our existing back-end has a login functionality. After successful active directory login the backend issues jwt tokens to the front-end where the tokens are stored.
I am trying to integrate this 'custom jwt authorization' into directus custom modules because I don't want to entertain two backend systems.

The problem: When I send the authorization-header with the custom jwt token the directus middleware seems to interfere and send a 401 with the following content: 

```
{
  "errors": [
    {
      "message": "Invalid user credentials.",
      "extensions": {
        "code": "INVALID_CREDENTIALS"
      }
    }
  ]
} 
```

Is there a way to circumvent this? Or should I put the token in the body?

**Comment by br41nslug** [comment]
When using the same authentication methods as Directus it needs to be a valid token known to Directus (`Authorization` header or `access_token` query param). A way around it as you suggested already is to use different naming or a different way of transporting the token. A few examples could be:
- `Token` header
- `?jwt=` parameter
- or sending it in the body

The names don't matter as long as they dont clash with the Directus core

**Reply by fusscreme** [reply]
Thank you very much for your answer :) 
I never heard of the `Token` header or the `?jwt=` parameter before. I'll do some googling.
Or maybe I phrased the question badly. I want to **receive** the token, not send it.

Nonetheless, your answer makes it clear that I have to try something different. I'll try to put the token in a custom header.
Thanks again.

**Reply by br41nslug** [reply]
They're not any sort of standard im aware of but just a new name i made up on the spot 😇 that doesnt conflict with the existing headers/params.
